### ISSUE TITLE
ENH: improve error reporting on windows when numpy fails to load

### ISSUE THREAD

**Issue by mattip** [numpy/numpy#13014]
We are getting many new issues where users cannot load numpy on windows. Some (most?, all?) of the errors seem to be due to Anaconda breaking the use of `os.environ['PATH']` but it is hard to separate the wheat from the chaff. What suggestions or help should we give to users so they can solve this on their own?

Which of these additional checks would be helpful, or would they be too error prone and fragile (where `.libs`) directory is the `extra_dll_dir` from [distutils] :
- Does the `_multiarray_umath.pyd` file exist?
- Does the `.libs` directory exist?
- Does the `.libs` directory contain a `*dll`
- Does `os.environ['PATH']` contain the `.libs` directory?
- Are there broken elements in `os.environ['PATH']`? (not sure whether this is an issue or not)

Any other ideas how to mitigate this? PR #12994 should solve this, but it seems to have stalled.

**Comment by tylerjereddy** [comment]
I may need to open a PR to demonstrate the "second best" approach for NumPy, which would be preferred by SciPy / ecosystem, based on discussion between Ralf & Steve Dower.

I'm a little nervous about making another `distutils` change after the `exec_command()` fallout, but I'll try the second approach so we have the two options on the table in active PRs at least.

**Comment by rgommers** [comment]
I don't think any of those questions are going to be helpful to be honest. The `os.environ['PATH']` ones can't be answered because the envvars get altered dynamically by conda Python (as you commented on in another issue). The `.libs` one you've now asked a bunch of users and most of them get confused. I'm also not sure that the answers will be useful.

If it continues like this, I think we will need a diagnostic script that users can run. To develop that, it would be useful if we could reproduce the PATH issue. I'm trying that right now, but no luck yet. Did you try/manage?

**Comment by tylerjereddy** [comment]
Not sure about PATH stuff--I thought the problem was that DLL resolution was bricked, so as long as we can load the DLL in memory numpy will import, even with scrambled PATH?

Are you suggesting that even if we put `DLL` in `core` (solution 1 in linked PR) or load it explicitly via absolute path in distutils (solution 2, maybe better for SciPy), changing the order of things in PATH can still break the numpy import?

**Comment by rgommers** [comment]
> Are you suggesting that even if we put `DLL` in `core` (solution 1 in linked PR) or load it explicitly via absolute path in distutils (solution 2, maybe better for SciPy), changing the order of things in PATH can still break the numpy import?

no, I'm not trying to discuss the solution here. we need a solution and release new wheels, but we also need to figure out how people are triggering this, and help them diagnose/fix. that's what this issue is about.

with diagnostic script I mean something like: tell the user to open Anaconda Prompt, and then execute:
```
conda list
pip list
... etc
```

**Comment by rgommers** [comment]
Part of the issue is also user education - if users are only hitting this because they install numpy with pip inside a conda env, then they're doing it fundamentally wrong anyway. other option is that numpy gets pulled in under the hood, and we may need to prevent/fix that somehow

**Comment by rgommers** [comment]
A big part of the problem here is created by the Anaconda defaults channel having an out of date NumPy - it's on 1.15.4 still. Many users will have an older pip version, and that will happily upgrade numpy when a user does `pip install -U somepackage-that-depends-on-numpy`. I'll ping the Anaconda guys on that.

**Comment by rgommers** [comment]
I went throught most of the issues now to get an impression. @mattip I think it's tricky to advise users to edit their PATH, especially putting things like our `.libs` on it. That's going to be error-prone and lead to more trouble down the line even if they figure it out.

> A big part of the problem here is created by the Anaconda defaults channel having an out of date NumPy - it's on 1.15.4 still.

An updated defaults isn't going to fix some cases I saw. E.g. users are creating an empty conda env and then doing `pip install tensorflow` straight away. That's still going to install numpy with pip.
